### PR TITLE
Refactoring of code for verify_data, precheck and setup functions

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Refactoring of code for verify_data, precheck and setup functions

Signed-off-by: Prajyot Parab <prajot.parab@ibm.com>

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -502,7 +506,8 @@ function precheck {
     SERVICE_INSTANCE_ID=$(grep "service_instance_id" $varfile | awk '{print $3}' | sed 's/"//g')
     debug_switch
     VAR_CLOUD_API_KEY=$(grep "ibmcloud_api_key" $varfile | awk '{print $3}' | sed 's/"//g')
-    [[ ! -z $VAR_CLOUD_API_KEY ]] && CLOUD_API_KEY=$VAR_CLOUD_API_KEY
+    [[ -n $VAR_CLOUD_API_KEY ]] && CLOUD_API_KEY=$VAR_CLOUD_API_KEY
+    VAR_PULL_SECRET=$(grep "pull_secret_file" "$varfile" | awk '{print $3}' | sed 's/"//g')
```
Comment (yussufsh) [inline deploy.sh:510]: Name it VAR_PULL_SECRET_FILE

File: deploy.sh [hunk 2]
```diff
@@ -601,7 +606,7 @@ function cluster_access_info {
     $($TF output bastion_ssh_command | sed 's/,.*//') -q -o StrictHostKeyChecking=no cat /root/openstack-upi/auth/kubeconfig > ./kubeconfig
     echo "To access the cluster on local system when using 'oc' run: 'export KUBECONFIG=$PWD/kubeconfig'"
     echo "Access the OpenShift web-console here: $($TF output web_console_url)"
-    echo "Login to the console with user: \"kubeadmin\", and password: \"$($($TF output bastion_ssh_command) -q -o StrictHostKeyChecking=no cat /root/openstack-upi/auth/kubeadmin-password)\""
+    echo "Login to the console with user: \"kubeadmin\", and password: \"$($($TF output bastion_ssh_command | sed 's/,.*//') -q -o StrictHostKeyChecking=no cat /root/openstack-upi/auth/kubeadmin-password)\""
```
Comment (yussufsh) [inline deploy.sh:609]: Can you do it everywhere we are using bastion ssh command?

File: deploy.sh [hunk 3]
```diff
@@ -831,13 +842,14 @@ function variables {
   fi
 
   if [ -s "./pull-secret.txt" ]; then
-    log "Found pull-secret.txt in current directory"
     cp -f pull-secret.txt ./"$ARTIFACTS_DIR"/data/
+    echo "pull_secret_file = \"$PWD/$ARTIFACTS_DIR/data/pull-secret.txt\"" >> $VAR_TEMPLATE
```
Comment (yussufsh) [inline deploy.sh:846]: Now I see it I think it's better not to copy and use it from CWD. Set the pull secret file var to PWD

File: deploy.sh [hunk 4]
```diff
@@ -831,13 +842,14 @@ function variables {
...
   else
     debug_switch
     question "Enter the pull-secret" "-sensitive"
     if [[ "${value}" != "" ]]; then
       echo "${value}" > ./"$ARTIFACTS_DIR"/data/pull-secret.txt
+      echo "pull_secret_file = \"$PWD/$ARTIFACTS_DIR/data/pull-secret.txt\"" >> $VAR_TEMPLATE
```
Comment (yussufsh) [inline deploy.sh:852]: This is fine.

File: deploy.sh [hunk 5]
```diff
@@ -882,11 +903,11 @@ function setup_terraform {
   EXT_PATH=$(which terraform 2> /dev/null || true)
 
   if [[ -f $TF && $($TF version | grep 'Terraform v0') == "Terraform ${TF_LATEST}" ]]; then
-    log "Terraform latest version already installed"
+    return
```
Comment (yussufsh) [inline deploy.sh:906]: So these messages won't be displayed during normal setup?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: deploy.sh (hunk 5 above)
Comment (Prajyot-Parab) [reply deploy.sh:905]: Yes, as the code will always ensure now when the script is ran everything is the latest.

File: deploy.sh [hunk 6]
```diff
@@ -466,13 +466,12 @@ function init_terraform {
...
-  elif [ -s "./pull-secret.txt" ]; then
-    log "Found pull-secret.txt in current directory"
-    cp -f pull-secret.txt ./"$ARTIFACTS_DIR"/data/
-  else
-    error "No pull-secret.txt file found in current directory"
+  if [ ! -s "$VAR_PULL_SECRET_FILE" ]; then
+    if [ -s "./pull-secret.txt" ]; then
+      cp -f ./pull-secret.txt ./"$ARTIFACTS_DIR"/data/
+    else
+      error "File pull-secret.txt file not found"
```
Comment (yussufsh) [inline deploy.sh:473]:
```
error "File pull-secret.txt not found"
```

**Review by yussufsh — APPROVED** [review]

**Pull request merged**